packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/useTeachingPopoverCarousel.ts

```diff
@@ -3,115 +3,136 @@ import {
   getIntrinsicElementProps,
   mergeCallbacks,
   slot,
-  useControllableState,
   useEventCallback,
+  useIsomorphicLayoutEffect,
+  useMergedRefs,
 } from '@fluentui/react-utilities';
-import type {
-  TeachingPopoverCarouselProps,
-  TeachingPopoverCarouselState,
-  TeachingPopoverPageCountChildRenderFunction,
-} from './TeachingPopoverCarousel.types';
+import type { TeachingPopoverCarouselProps, TeachingPopoverCarouselState } from './TeachingPopoverCarousel.types';
 import { Button } from '@fluentui/react-button';
-import { useState } from 'react';
+import { useSyncExternalStore } from 'use-sync-external-store/shim';
 import { usePopoverContext_unstable } from '@fluentui/react-popover';
 import { TeachingPopoverCarouselNav } from '../TeachingPopoverCarouselNav/TeachingPopoverCarouselNav';
+import { useCarousel_unstable } from './Carousel/Carousel';
 
 export const useTeachingPopoverCarousel_unstable = (
   props: TeachingPopoverCarouselProps,
   ref: React.Ref<HTMLDivElement>,
 ): TeachingPopoverCarouselState => {
-  const { layout = 'centered', paginationType = 'icon', strings, onPageChange, onFinish } = props;
+  const {
+    layout = 'centered',
+    paginationType = 'icon',
+    onPageChange,
+    onFinish,
+    currentPage,
+    defaultCurrentPage,
+  } = props;
+  const { carousel, carouselWalker, carouselRef } = useCarousel_unstable();
+
+  const { store, setValue, setIndex, currentIndex: _currentPage, totalPages: _totalPages, value } = carousel;
 
   const appearance = usePopoverContext_unstable(context => context.appearance);
-  const triggerRef = usePopoverContext_unstable(context => context.triggerRef);
   const toggleOpen = usePopoverContext_unstable(context => context.toggleOpen);
 
-  const reactChildArray = React.Children.toArray(props.children);
+  const values = useSyncExternalStore(store.subscribe, () => store.getSnapshot());
+  const totalPages = values.length;
```

Comment:
This component should not consume state from `useSyncExternalStore()` as it will make the source of updates, only child components should use that subscription